tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6487,6 +6487,9 @@ bool HLSLExternalSource::MatchArguments(
     }
   }
 
+  std::string profile = m_sema->getLangOpts().HLSLProfile;
+  const ShaderModel *SM = hlsl::ShaderModel::GetByName(profile.c_str());
```

Comment:
I wonder if we couldn't have a LangOpt for HLSLLongVectorSupport, initialized based on the shader model instead?